### PR TITLE
Add `with()` method to support external ServiceInstanceListSupplier(#988)

### PR DESCRIPTION
**Pull request by chenzhiguo**
Add `with()` method to support external `ServiceInstanceListSupplier` extension(#988)

### CODE REVIEW

**Comment by OlgaMaciaszek** [comment]
Fixes gh-988.

File: spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/ServiceInstanceListSupplierBuilder.java [hunk 1]
```diff
@@ -248,6 +248,13 @@ public ServiceInstanceListSupplierBuilder withHints() {
 		return this;
 	}
 
+	public ServiceInstanceListSupplierBuilder with(DelegateCreator delegateCreator) {
```
Comment (OlgaMaciaszek) [inline spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/ServiceInstanceListSupplierBuilder.java:251]: Please add javadoc for method.

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
@chenzhiguo Thanks for submitting this PR. The change looks good. Only cosmetic javadoc changes are needed: 
- I've added a comment - please address
- please add `@author` tag with your full name to the class javadoc.

File: spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/ServiceInstanceListSupplierBuilder.java (hunk 1 above)
Comment (chenzhiguo) [reply spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/ServiceInstanceListSupplierBuilder.java:251]: OK, I will add it soon.

Comment (chenzhiguo) [reply spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/ServiceInstanceListSupplierBuilder.java:258]: @OlgaMaciaszek Please check it, meet the requirements?

**Comment by OlgaMaciaszek** [comment]
@chenzhiguo 
> [ERROR] /home/appuser/project/spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/ServiceInstanceListSupplierBuilder.java:252: First sentence should end with a period. [JavadocStyle]

Basically, you need to add a full stop at the end of the first javadoc sentence (in line 254).

**Review by OlgaMaciaszek — APPROVED** [review]
LGTM.

**Pull request merged**